### PR TITLE
Add a `.gte()` method for checking if a version is greater than or equal to another

### PR DESCRIPTION
**Pull request by LucianBuzzo**
Connects to #15 

change-type: minor

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -150,3 +152,54 @@ export const major = (version: string | null): number | null => {
...
+
+	if (!isAValid && isBValid) {
+		return false;
+	}
+
+	if (isAValid && !isBValid) {
+		return true;
+	}
+
+	return semver.gte(versionA, versionB);
```
Comment (pimterry) [inline src/index.ts]: This has inconsistent behaviour with `compare`, which takes `dev` and `rev` suffixes into account. I think what should probably happen is that `gte` should be equivalent to `compare(a, b) >= 0`. Is there any reason those should give different results?

Comment (LucianBuzzo) [inline src/index.ts]: Yes, I missed this. These comparison methods (lt, gt, gte etc) should be using compare internally. I'll add some test cases for dev and rev as well 👍

**Comment by LucianBuzzo** [comment]
@pimterry I've added comprehensive testing for the `.compare()` function and fixed the implementation as it was previously performing a right compare, meaning the results were the opposite of what you would expect. The `.gte()` function now wraps `.compare()`. I'll propagate these changes to the other PRs once this one is merged.

File: src/index.ts [hunk 2]
```diff
@@ -189,3 +197,29 @@ export const prerelease = (version: string | null) => {
...
+ *
+ * @param {string|null} versionA - The version string to compare against
+ * @param {string|null} versionB - The version string to compare to versionA
+ *
+ * @returns {boolean} - true if versionA is greater than or equal to versionB, otherwise false.
+ *
+ * @example
+ * resinSemver.gte('2.0.5', '1.16.0'); //--> true
+ *
+ * resinSemver.gte('Resin OS 2.0.5', 'Resin OS 2.0.2+rev2'); //--> false
```
Comment (pimterry) [inline src/index.ts]: I think this example is wrong - this will (correctly) return `true`

**Pull request merged**